server/controllers/finance/journal/index.js

```diff
@@ -36,6 +40,64 @@ exports.journalEntryList = journalEntryList;
 
 exports.editTransaction = editTransaction;
 exports.count = count;
+exports.refenreceTransaction=refenreceTransaction;
+
+
+
+
+
+//loading identifiers keyies, used for defining the table name
+const identifiersIndex = {};
+function indexIdentifiers() {
+  _.forEach(identifiers, (entity) => {
+    identifiersIndex[entity.key] = entity;
+  });
+}
+
+// API GET /journal/refenrece_transaction/:codeRef/:language'
+
+//This function render a report in the browser
+//It takes a transaction reference code as paramter and the language
+//The reference code is a combination of table_key.project_abbr.reference
+//The table name is variable, it can be :invoice, cash or voucher
+
+function refenreceTransaction(req,res,next){
+   
+  //loading identifiers keyies, used for defining the table name
+  indexIdentifiers();
+
+  //reports urls
+  var reportsTypes={
```

Comment:
This map already exists in the `identifiers.js` object that you import - it would be best not to duplicate this behaviour.

I think it would be best to add these as `redirectPath` attributes to the already existing object. 

This code could then run without defining report types or indexing them.